mulog-kafka/test/com/brunobonacci/mulog/publishers/kafka_test.clj

```diff
@@ -0,0 +1,50 @@
+(ns com.brunobonacci.mulog.publishers.kafka-test
+  (:require [midje.sweet :refer :all]
+            [com.brunobonacci.mulog :as mu]
+            [com.brunobonacci.mulog.publishers.container.kafka-setup :as kafka-setup]
+            [clojure.core.async :as async]
+            [jsonista.core :as json]
+            [org.slf4j.impl.mulog]))
+
+(def ^:private ^:const TOPIC "mulog")
+
+(def ^:private json-mapper (json/object-mapper {:decode-key-fn true}))
+
+(defn- <!!? [chan milliseconds]
+  (let [timeout (async/timeout milliseconds)
+        [value _] (async/alts!! [chan timeout])]
+    value))
+
+(let [stop-console-pub (atom nil)
+      stop-kafka-pub   (atom nil)
+      kafka-container  (atom nil)
+      kafka-consumer   (atom nil)]
+  (with-state-changes [(before :facts
+                               (do
+                                 (reset! stop-console-pub (mu/start-publisher! {:type :console}))
+                                 (reset! kafka-container (kafka-setup/start-container!))
+                                 (reset! stop-kafka-pub (mu/start-publisher! {:type      :kafka
+                                                                              :kafka     {:bootstrap.servers (kafka-setup/get-bootstrap-servers @kafka-container)}
+                                                                              :topic     TOPIC
```

Comment:
To isolate each test on the same cluster we could use different topic, like:
`(str "mulog-" (f/snowflake))`

so each test will run on a separate topic thus isolating results.